### ISSUE TITLE
Commands need `--help`/`-h`

### ISSUE THREAD

**Issue by wcooley** [berdario/pew#138]
A few commands have `argparse` parsers and so they accept a `--help` parameter, but most of them don't. Even if the arguments are trivial (or non-existent), I expect `--help` to tell me so, rather than having switch contexts to a browser to review documentation of basic operation.

This might be a good use of an 3rd-party command-line parser, like [`click`](http://click.pocoo.org/6/)--it would add `--help` _and_ take care of registering the subcommands.

**Comment by erinxocon** [comment]
You are correct, I will had help messages to make the cli more user friendly.  @berdario any opinions on using click as a cli parser?

**Comment by berdario** [comment]
I'm :+1: on making the help messages consistent (this was not done just due to laziness :/ )

I'm :-1:  on using Click

I cannot find where this discussion originally happened, but it was in the context of discussing https://github.com/ryanhiebert/pew2

basically, Click (at least at the time, maybe now it changed?) has a quite cumbersome api, that dissuades from reusing the common parts in different subcommands parsers

see this [gist](https://gist.github.com/berdario/9e748f154f0473e1f5e2)

Ideally, in Python I'd like to use/reimplement something like [optparse-applicative](https://hackage.haskell.org/package/optparse-applicative#readme)

**Comment by erinxocon** [comment]
@berdario If you take a look at the `pipenv` project by @kennethreitz; we use click over there and runs well.  @kennethreitz any opinions on the click api?

**Comment by berdario** [comment]
Yeah, I had a look at the argument parsing of pipenv, and the symptom I was talking about is quite visible, namely:

     @click.option('--three/--two', is_flag=True, default=None, help="Use Python 3/2 when creating virtualenv.")

And such options are repeated times and times again

**Comment by erinxocon** [comment]
That is true, however it's only repeated if you want to have the same arguments for all commands, which is necessary in pipenv.  It could get to be a bit hairy with pew as well.  I'm down to try out optparse, or a smaller library.  Click is a rather high level api for cli apps and as such comes with some overhead.

**Comment by berdario** [comment]
Well, the point is that several commands have a common subset of options.

You mention optparse, be aware that what I was talking about is not Pyhton's optparse (which is an outdated library).
